modules/surface_matching/src/icp.cpp

```diff
@@ -540,12 +540,22 @@ int ICP::registerModelToScene(const Mat& srcPC, const Mat& dstPC, double& residu
 // source point clouds are assumed to contain their normals
 int ICP::registerModelToScene(const Mat& srcPC, const Mat& dstPC, std::vector<Pose3DPtr>& poses)
 {
-  for (size_t i=0; i<poses.size(); i++)
+  #if defined _OPENMP
```

Comment:
@sovrasov I agree that using `cv::parallel_for` seems more wise, and I would consider that in my rewrite version. However, the rest of the module is using OpenMP for now, so I prefer to remain consistent.

As for the `omp critical` part, although it does not seem to be needed, it does not harm the execution as well. Heavy processes are already done before the critical part, and it is just a kind of protective practice. Anyway, I am removing it since you do not like such practice.